linux-installer.sh

```diff
@@ -91,44 +109,85 @@ has_wget() {
 # Stepping:            9
 # CPU MHz:             2900.000
 has_lscpu() {
-  echo "TODO"
+   if [[ -n "$(command -v lscpu)" ]]; then
+      return 0
+   fi
+   return 1
 }
 
 # Check if we have /proc/cpuinfo
 has_proc_cpuinfo() {
-  echo "TODO"
+   if [[ -e /proc/cpuinfo ]]; then
+      return 0
+   fi
+   return 1
+}
+
+check_ark_intel() {
+   productUrl=$(curl "https://ark.intel.com/search/AutoComplete?term=${cpuModel}" | sed -n 's/.*\"quickUrl\":\"\(.*\)\".*/\1/p')
+   tmpCpuType=$(curl --silent "https://ark.intel.com${productUrl}" | sed -n 's/.*Products formerly \(.*\)<.*/\1/p')
+   CPU_TYPE=$(echo $tmpCpuType | sed 's/ //g' | awk '{print tolower($0)}')
 }
 
 # Check CPU type
 check_cpu_type() {
-  echo "TODO"
-  # If we have lscpu, use the following command:
-
-  # lscpu | sed -nr '/Model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3
-  # Example: i7-7820HQ
-
-  # If we have /proc/cpuinfo, use the following command:
-  # cat /proc/cpuinfo | sed -nr '/model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3 | head -1
-  # Example: i7-7820HQ
+   CPU_TYPE="core2"
+
+   if has_lscpu; then
+      cpuModel=$(lscpu | sed -nr '/Model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3)
+      check_ark_intel
+   elif has_proc_cpuinfo; then
+      cpuModel=$(cat /proc/cpuinfo | sed -nr '/model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3 | head -1)
```

Comment:
We need to do some stupid bounds checking because there's CPUs that have non-conventional names. For example, my CPU - Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz = `i7-7820HQ` but `Intel(R) Xeon(R) CPU E3-1220` = `CPU`.

Even bigger problem is:

```
  # Intel(R) Xeon(R) CPU E3-1220 @ 2.5GHz
  # Intel(R) Xeon(R) CPU E3-1220 v2 @ 2.5GHz
  # Intel(R) Xeon(R) CPU E3-1220 v3 @ 2.5GHz
  # Intel(R) Xeon(R) CPU E3-1220 0 @ 2.5GHz
```
`0` and `v2` and `v3` are all different microarchitectures... what?